crypto/src/mls/credential/mod.rs

```diff
@@ -332,86 +332,89 @@ pub mod tests {
     #[wasm_bindgen_test]
     async fn should_not_fail_but_degrade_when_basic_joins(case: TestCase) {
         if case.is_x509() {
-            let mut x509_test_chain = X509TestChain::init_empty(case.signature_scheme());
-
-            let (alice_identifier, _) = x509_test_chain.issue_simple_certificate_bundle("alice", None);
-            let (bob_identifier, _) = x509_test_chain.issue_simple_certificate_bundle("bob", None);
-
-            // this should work since the certificate is not yet expired
-            let (mut alice_central, mut bob_central, id) =
-                try_talk(&case, Some(&x509_test_chain), alice_identifier, bob_identifier)
-                    .await
-                    .unwrap();
-
-            assert_eq!(
-                alice_central.e2ei_conversation_state(&id).await.unwrap(),
-                E2eiConversationState::Verified
-            );
-
-            assert_eq!(
-                bob_central.e2ei_conversation_state(&id).await.unwrap(),
-                E2eiConversationState::Verified
-            );
-
-            alice_central.try_talk_to(&id, &mut bob_central).await.unwrap();
-            assert_eq!(
-                alice_central.e2ei_conversation_state(&id).await.unwrap(),
-                E2eiConversationState::Verified
-            );
-
-            assert_eq!(
-                bob_central.e2ei_conversation_state(&id).await.unwrap(),
-                E2eiConversationState::Verified
-            );
+            Box::pin(async {
```

Comment:
Nitpick: we could check if the case is not using x509 credentials and then do an early return, which would save us one level of indentation.

Alternatively, we could create a new test fixture, similar to `all_cred_cipher`, e.g. `all_cert_cipher`, which would use only x509 credentials. See [this](https://github.com/wireapp/core-crypto/blob/31e1d63d9ea412624ca469a9c4f75fb6270029a2/crypto/src/test_utils/fixtures.rs#L97). Such a test fixture would allow us to get rid of the above checks in a bunch of other places, too.